ext/native.go

```diff
@@ -188,16 +188,17 @@ func (tp *nativeTypeProvider) FindStructFieldType(typeName, fieldName string) (*
 	if !ok {
 		return nil, false
 	}
+	refFieldName := refField.Name
```

Comment:
I'm not sure I follow how the input field name differs from the `refField.Name` value in this case. Could you add a test to illustrate the change in behavior?